src/hardware/sblaster.cpp

```diff
@@ -588,8 +588,12 @@ static void DSP_FlushData()
 static double last_dma_callback = 0.0;
 
 static void DSP_DMA_CallBack(DmaChannel * chan, DMAEvent event) {
-	if (chan!=sb.dma.chan || event==DMA_REACHED_TC) return;
-	else if (event==DMA_MASKED) {
+	assert(chan && chan == sb.dma.chan);
+
```

Comment:
If this fails that can only mean a DOSBox side programming error, right? The more I think about it, invalid behaviour that can happen on real hardware too should never be asserted but handled.

Reply:
Yup, this is just coding sanity check and to document expectations (which should be 100% guaranteed), to make sure order-of-operations are being handled properly.

Right; asserts should never be used to validate user-side data, and instead, they should represent expected truths about the state of the code (and can act to catch broken program flow before it derails further into something harder to debug).

Reply:
Yeah, the original devs didn't use asserts, so when reading/refactoring code it's hard to separate expectations versus nominal pathway handling.

In this case, DMA callbacks are called by DMA channel objects, and pass themselves into the function as the `* chan` pointer.  And there's no way a `nullptr` DMA channel will making this callback.